app/src/main/java/com/example/keniphhomemaintenance/maintenance_screen/MaintenanceView.kt

```diff
@@ -0,0 +1,252 @@
+                }
+            ) {
+                Icon(Icons.Default.Add, contentDescription = "Button to add a maintenance item")
+            }
+        },
+        floatingActionButtonPosition = FabPosition.Center
+    ){
+            ShowDialog(isDialogOpen)
+    }
+
+    MaintenanceItemsList(maintenanceList)
+
+}
+
+
+
+@Composable
+fun MaintenanceCard(maintenanceItem: MaintenanceItem) {
+    Spacer(modifier = Modifier.width(dimensionResource(R.dimen.maintenance_card_start_padding)))
+    Row(
+        modifier = Modifier
+            .padding(vertical = dimensionResource(R.dimen.maintenance_card_row_vertical_padding))
+            .padding(start = dimensionResource(R.dimen.maintenance_card_row_start_padding))
+            .fillMaxWidth(),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Image(
+            painter = painterResource(R.drawable.ic_baseline_build_24), 
+            contentDescription = "Maintenance Item Type",
+            modifier = Modifier
+                .size(dimensionResource(R.dimen.maintenance_image_size))
+                .clip(CircleShape)
+        )
+        
+        Spacer(modifier = Modifier.width(dimensionResource(R.dimen.maintenance_image_end_padding)))
+
+        Column {
+            Text(
+                text = maintenanceItem.title,
+                color = MaterialTheme.colors.primaryVariant,
+                style = MaterialTheme.typography.h5
+            )
+            Spacer(modifier = Modifier.height(dimensionResource(R.dimen.maintenance_title_and_due_date_padding)))
+            Text(
+                text = maintenanceItem.dueDate,
+                color = Color.Black,
+                style = MaterialTheme.typography.body2
+            )
+            Text(
+                text = maintenanceItem.location,
+                color = Color.Black,
+                style = MaterialTheme.typography.body2
+
+            )
+        }
+    }
+}
+
+@Composable
+fun MaintenanceItemsList(maintenanceItems: List<MaintenanceItem>) {
+    LazyColumn(
+        modifier = Modifier
+            .fillMaxHeight()
+            .padding(bottom = dimensionResource(R.dimen.maintenance_item_list_bottom_padding))
+    ) {
+        items(maintenanceItems) { maintenanceItems ->
+            MaintenanceCard(maintenanceItems)
+        }
+    }
+}
+
+@Composable
+fun ShowDialog(isDialogOpen: MutableState<Boolean>) {
+    val itemTitle = remember { mutableStateOf("") }
+    val dueDate = remember { mutableStateOf("") }
+    val location = remember { mutableStateOf("") }
+
+    if (isDialogOpen.value) {
+        Dialog(
+            onDismissRequest = { isDialogOpen.value = false }) {
+            Surface(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(vertical = dimensionResource(R.dimen.maintenance_item_dialog_vertical_padding)),
+              shape = RoundedCornerShape(dimensionResource(R.dimen.maintenance_item_dialog_corner_radius)),
```

Comment:
weird spacing thing; there's a keyboard shortcut for fixing spacing in your code. Highlight anything you want to autofix --> ctrl+alt+L

Reply:
@NickPerozzi for the neat keyboard shortcut

Reply:
LeftCtrl+A to RightCtrl+RightAlt+L is my favorite speedrun strat